include/swift/AST/Expr.h

```diff
@@ -603,6 +603,10 @@ class alignas(8) Expr {
   /// a builtin operator.
   bool isInfixOperator() const;
 
+  /// Returns true if this expression needs to be wrapped in parens when
+  /// used inside of a postfix operator like "!" or "?".
+  bool needsParensInsidePostfixOperator() const;
```

Comment:
This is pretty much the same as `canAppendCallParentheses`. Maybe merge them as `canAppendPostfixExpression` or something?